physics/continuous_trajectory.hpp

```diff
@@ -80,16 +75,28 @@ class ContinuousTrajectory {
   // Removes all data for times strictly less than |time|.
   void ForgetBefore(Instant const& time);
 
-  // Evaluates the trajectory at the given |time|, which must be in
-  // [t_min(), t_max()].  The |hint| may be used to speed up evaluation
-  // in increasing time order.  It may be a nullptr (in which case no speed-up
-  // takes place).
-  virtual Position<Frame> EvaluatePosition(Instant const& time,
-                                           Hint* hint) const;
-  virtual Velocity<Frame> EvaluateVelocity(Instant const& time,
-                                           Hint* hint) const;
-  virtual DegreesOfFreedom<Frame> EvaluateDegreesOfFreedom(Instant const& time,
-                                                           Hint* hint) const;
+  // Implementation of the interface |Trajectory|.
+
+  using TrajectoryHint = Trajectory<Frame>::Hint;
```

Comment:
I don't like this name, I'd rather have `Trajectory<Frame>::Hint` in the few places where it's used.